psamm/commands/completepath.py

```diff
@@ -0,0 +1,280 @@
+            help='Do not include implicit sinks when gap-filling')
+        parser.add_argument(
+            '--allow-bounds-expansion', action='store_true',
+            help=('Allow GapFill to propose expansion of flux bounds. This'
+                  ' includes turning irreversible reactions reversible.'))
+        parser.add_argument(
+            '--gapfill', action='store_true',
+            help='Run the gapfill algorithm on the model with separated '
+                 'sink and source reactions in each compatment.')
+        super(CompletePathCommand, cls).init_parser(parser)
+
+    def run(self):
+        """Run CompletePath command"""
+
+        # Load compound information
+        def compound_name(id):
+            if id not in self._model.compounds:
+                return id
+            return self._model.compounds[id].properties.get('name', id)
+
+        # Reaction gene information
+        def reaction_genes_string(id):
+            if id not in self._model.reactions:
+                return 'No Gene'
+            return self._model.reactions[id].properties.get('genes', '')
+
+        # Calculate penalty if penalty file exists
+        penalties = {}
+        if self._args.penalty is not None:
+            for line in self._args.penalty:
+                line, _, comment = line.partition('#')
+                line = line.strip()
+                if line == '':
+                    continue
+                rxnid, penalty = line.split(None, 1)
+                penalties[rxnid] = float(penalty)
+
+        core = set(self._mm.reactions)
+
+        solver = self._get_solver(integer=True)
+        default_comp = self._model.default_compartment
+        epsilon = self._args.epsilon
+        v_max = float(self._model.default_flux_limit)
+
+        blocked = set()
+        for compound in self._args.compound:
+            if compound.compartment is None:
+                compound = compound.in_compartment(default_comp)
+            blocked.add(compound)
+
+        if len(blocked) > 0:
+            logger.info('Unblocking compounds: {}...'.format(
+                ', '.join(text_type(c) for c in sorted(blocked))))
+        else:
+            logger.info(
+                'Unblocking all compounds in model. Use --compound option to'
+                ' unblock specific compounds.')
+            blocked = set(self._mm.compounds)
+
+        exclude = set()
+        if self._model.biomass_reaction is not None:
+            exclude.add(self._model.biomass_reaction)
+
+        # Add exchange and transport reactions to database
+        model_complete, weights = create_extended_sink_source_model(
+            self._model,
+            db_penalty=self._args.db_penalty,
+            source_penalty=self._args.source_penalty,
+            sink_penalty=self._args.sink_penalty,
+            tp_penalty=self._args.tp_penalty,
+            penalties=penalties)
+
+        implicit_sinks = not self._args.implicit_sinks
```

Comment:
This should be `... = self._args_implicit_sinks`? I seems that the `not` accidentally reverses the logic.